kotlin/samples/containers/common/src/main/java/com/squareup/sample/container/masterdetail/MasterDetailScreen.kt

```diff
@@ -23,11 +23,62 @@ import com.squareup.workflow.ui.BackStackScreen
  * Containers may choose to display both children side by side in a split view, or concatenate them
  * (master + detail) in a single pane.
  *
- * @param selectDefault optional function that a split view container make call to request
+ * @param selectDefault optional function that a split view container may call to request
  * that a selection be made to fill a null [detailRendering].
  */
-data class MasterDetailScreen(
+class MasterDetailScreen private constructor(
   val masterRendering: BackStackScreen<Any>,
   val detailRendering: BackStackScreen<Any>? = null,
   val selectDefault: (() -> Unit)? = null
-)
+) {
+  constructor(
+    masterRendering: BackStackScreen<Any>,
+    detailRendering: BackStackScreen<Any>
+  ) : this(masterRendering, detailRendering, null)
+
+  /**
+   * @param selectDefault optional function that a split view container may call to request
+   * that a selection be made to fill a null [detailRendering].
+   */
+  constructor(
+    masterRendering: BackStackScreen<Any>,
+    selectDefault: (() -> Unit)? = null
+  ) : this(masterRendering, null, selectDefault)
+
+  operator fun component1(): BackStackScreen<Any> = masterRendering
+  operator fun component2(): BackStackScreen<Any>? = detailRendering
+
+  operator fun plus(other: MasterDetailScreen): MasterDetailScreen {
```

Comment:
Needs kdoc.

Reply:
🐮 🔔

Reply:
Needs unit testing too.

Reply:
```
  /**
   * Returns a new [MasterDetailScreen] appending the [masterRendering] and
   * [detailRendering] of [other] to those of the receiver. If the new screen's
   * [detailRendering] is `null`, it will have the [selectDefault] function of [other].
   */
  fun combine(other: MasterDetailScreen): MasterDetailScreen {
```